src/framework/theme/components/search/search.component.ts

```diff
@@ -239,14 +243,17 @@ export class NgaSearchComponent implements OnInit, AfterViewInit, OnDestroy {
     const fieldComponentObservable = this.searchType === NgaSearchFieldComponent.TYPE_SIMPLE_SEARCH ?
       this.createAttachedSearch(NgaSearchFieldComponent)
       : this.themeService.appendToLayoutTop(NgaSearchFieldComponent);
-    fieldComponentObservable.subscribe((componentRef: ComponentRef<any>) => {
+    this.createFieldSubscription = fieldComponentObservable.subscribe((componentRef: ComponentRef<any>) => {
       if (componentRef) {
         this.connectToSearchField(componentRef);
       }
     });
   }
 
   ngOnDestroy() {
-    this.themeService.clearLayoutTop();
+    this.activateSearchSubscription.unsubscribe();
+    this.deactivateSearchSubscription.unsubscribe();
+    this.createFieldSubscription.unsubscribe();
+    this.searchFieldComponentRef.destroy();
```

Comment:
Yep, right, each search comopnent destroy his field, instead of clear all layout-top.